python/ray/experimental/workflow/serialization.py

```diff
@@ -0,0 +1,94 @@
+import asyncio
+from dataclasses import dataclass
+from enum import Enum
+import ray
+from ray.types import ObjectRef
+from ray.experimental.workflow.common import calculate_identifier
+from ray.experimental.workflow.storage import Storage
+from ray.experimental.workflow import workflow_storage
+from typing import Any, Dict, List, Optional, Tuple, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ray.actor import ActorHandle
+
+MANAGEMENT_ACTOR_NAME = "StorageManagementActor"
+MANAGEMENT_ACTOR_NAMESPACE = "workflow"
+
+
+def init_manager(storage: Storage) -> None:
+    handle = Manager.options(
+        name=MANAGEMENT_ACTOR_NAME,
+        namespace=MANAGEMENT_ACTOR_NAMESPACE,
+        lifetime="detached").remote(storage)
+    ray.get(handle.ping.remote())
+
+
+def get_manager() -> "ActorHandle":
+    return ray.get_actor(
+        MANAGEMENT_ACTOR_NAME, namespace=MANAGEMENT_ACTOR_NAMESPACE)
+
+
+class UploadState(Enum):
+    IN_PROGRESS = 1
+    FINISHED = 2
+
+
+@dataclass
+class Upload:
+    state: UploadState
+    identifier_ref: ObjectRef[str]
+    upload_task: Optional[ray.ObjectRef]
+
+
+@ray.remote
+def _put_helper(identifier: str, obj: Any,
+                wf_storage: "workflow_storage.WorkflowStorage") -> None:
+    paths = wf_storage._key_obj_id(identifier)
```

Comment:
Put assert here to make sure obj is not `ObjectRef`

Reply:
hmmm we can add that assert for now since we aren't supporting nested object refs, but in general its possible `foo.remote(ray.put(ray.put(x))` for example.

Reply:
Right. I think right now, if obj is object ref, it just won't work. We can relax the check here once we support that.
My concern here is that if the user puts nested obj refs into the code, it won't work and we don't raise error. That'll be bad.